File: backend/internal/model/online_user.go
package model

import "time"

type OnlineUser struct {
	UserID    int64     `json:"user_id"`
	Username  string    `json:"username"`
	Nickname  string    `json:"nickname"`
	Email     string    `json:"email"`
	Role      string    `json:"role"`
	TenantID  int64     `json:"tenant_id"`
	IP        string    `json:"ip"`
	UserAgent string    `json:"user_agent"`
	LoginAt   time.Time `json:"login_at"`
	LastSeen  time.Time `json:"last_seen"`
}

File: web/src/api/onlineUser.ts
import request, { type ApiResponse } from './request'

export interface OnlineUser {
  user_id: number
  username: string
  nickname: string
  email: string
  role: string
  tenant_id: number
  ip: string
  user_agent: string
  login_at: string
  last_seen: string
}

const onlineUserApi = {
  getOnlineUsers(): Promise<ApiResponse<OnlineUser[]>> {
    return request.get('/online-users')
  },

  forceLogout(userId: number): Promise<ApiResponse<null>> {
    return request.post('/online-users/force-logout', { user_id: userId })
  },

  getOnlineCount(): Promise<ApiResponse<{ count: number }>> {
    return request.get('/online-users/count')
  }
}

export default onlineUserApi
